webpack/components/Table/TableWrapper.js

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+} from '@patternfly/react-table';
+import {
+  Pagination,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { STATUS } from 'foremanReact/constants';
+import { useForemanSettings } from 'foremanReact/Root/Context/ForemanContext';
+
+import EmptyStateMessage from './EmptyStateMessage';
+import Loading from './Loading';
+
+/* Patternfly 4 table wrapper */
+const TableWrapper = ({
+  status, cells, rows, error, emptyTitle, emptyBody, metadata, fetchItems, ...extraTableProps
+}) => {
+  const dispatch = useDispatch();
+  const { foremanPerPage = 20 } = useForemanSettings();
+  // setting pagination to local state so it doesn't disappear when page reloads
+  const [perPage, setPerPage] = useState(foremanPerPage);
+  const [page, setPage] = useState(1);
+  const [total, setTotal] = useState(0);
```

Comment:
Could the pagination-related stuff be extracted into a `usePagination()` hook?  It would be awesome if we could reuse that across different tables, etc.